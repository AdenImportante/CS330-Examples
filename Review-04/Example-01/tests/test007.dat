#! /bin/sh

# Test Armour::read and Armour::display
# formatting must match

cat <<EOF > test007.cpp

#include <iostream>
#include <iomanip>
#include <sstream>

#include "Armour.h"

using std::cout;
using std::istringstream;

int main(){
    Item *armour = new Armour(); 

    istringstream ins( "Boots Iron 1 2 FeatherFalling 3 Water" );

    armour->read( ins );    
    armour->display( cout );  

    return 0;
}
EOF

g++ -c test007.cpp 
g++ -o test007 test007.o Armour.o Item.o

./test007