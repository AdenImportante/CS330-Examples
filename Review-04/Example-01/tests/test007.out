  Nme: Boots
  Dur: 1
  Def: 2
  Mtl: Iron
  Mdr: FeatherFalling (Lvl 3)
  Emt: Water
